scripts/run-docker.sh

```diff
@@ -54,21 +68,26 @@ $SUDO docker start $CONTAINER_NAME >/dev/null 2>&1 || {
 		--tty \
 		$TERMUX_BUILDER_IMAGE_NAME
 	if [ "$UNAME" != Darwin ]; then
-		if [ $(id -u) -ne 1001 -a $(id -u) -ne 0 ]; then
+		REPO_UID="$(stat -c %u $REPOROOT)"
+		REPO_GID="$(stat -c %g $REPOROOT)"
+		if [ "$REPO_UID" -eq "0" ]; then
+			echo "Warning, the repository is cloned by root. Because of this, some script functions will not work in container."
+		elif [ $REPO_UID -ne 1001 ]; then
 			echo "Changed builder uid/gid... (this may take a while)"
-			$SUDO docker exec $DOCKER_TTY $CONTAINER_NAME sudo chown -R $(id -u) $CONTAINER_HOME_DIR
-			$SUDO docker exec $DOCKER_TTY $CONTAINER_NAME sudo chown -R $(id -u) /data
-			$SUDO docker exec $DOCKER_TTY $CONTAINER_NAME sudo usermod -u $(id -u) builder
-			$SUDO docker exec $DOCKER_TTY $CONTAINER_NAME sudo groupmod -g $(id -g) builder
+			$SUDO docker exec $DOCKER_TTY $CONTAINER_NAME sudo chown -R $REPO_UID $CONTAINER_HOME_DIR
+			$SUDO docker exec $DOCKER_TTY $CONTAINER_NAME sudo chown -R $REPO_UID /data
+			$SUDO docker exec $DOCKER_TTY $CONTAINER_NAME sudo usermod -u $REPO_UID builder
+			$SUDO docker exec $DOCKER_TTY $CONTAINER_NAME sudo groupmod -g $REPO_GID builder
 		fi
 	fi
 }
 
-# Set traps to ensure that the process started with docker exec and all its children are killed. 
-. "$TERMUX_SCRIPTDIR/scripts/utils/docker/docker.sh"; docker__setup_docker_exec_traps
-
-if [ "$#" -eq "0" ]; then
-	set -- bash
+if [ "$STOP_CONTAINER" = "true" ]; then
+	echo "Stopping container '$CONTAINER_NAME'..."
+	$SUDO docker stop $CONTAINER_NAME >/dev/null 2>&1
+	$SUDO docker rm $CONTAINER_NAME >/dev/null 2>&1
```

Comment:
`docker rm` will also remove the container entirely and it will then need to be redownloaded to start it again. And such functionality should ideally exist in a separate script instead of in a script named `run-docker.sh`. The `run-docker.sh` script could first be moved to a `termux-docker.sh` script and committed. Then a new script `run-docker.sh` can be created that just runs `termux-docker.sh start`. The `termux-docker.sh` script can then have the actual code for `start`, `stop`, `remove`/`prune` commands. This way should preserve git history of the old `run-docker.sh` script.

The stop command doesn't need to run any other logic anyways and just needs container name, there is needless mixing of logic going on currently.

Reply:
> docker rm will also remove the container entirely and it will then need to be redownloaded to start it again.

If by redownload you mean downloading the docker image from the repo, that won't happen. This command removes the container that was running, but the original image for the container that is loaded on the computer is not touched.

> The stop command doesn't need to run any other logic anyways and just needs container name, there is needless mixing of logic going on currently.

If by mixing logic you mean running `docker stop` with `docker rm`, then this is done specifically so that you can "reboot" (i.e. start the container again with changes in the environment if there are any) the container.